### PR TITLE
MAKE-521: ftdc2csv tools

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: operations/ftdc.go [hunk 1]
```diff
@@ -278,15 +311,131 @@ func fromBSON() cli.Command {
...
+			if outputPath == "" {
+				outputFile = os.Stdout
+			} else {
+				if _, err = os.Stat(outputPath); !os.IsNotExist(err) {
+					return errors.Errorf("cannot write ftdc to '%s', file already exists", outputPath)
+				}
+
+				outputFile, err = os.Create(outputPath)
+				if err != nil {
+					return errors.Wrapf(err, "problem opening flie '%s'", outputPath)
```
Comment (julianedwards) [inline operations/ftdc.go]: *file

File: operations/ftdc.go [hunk 2]
```diff
@@ -278,15 +311,131 @@ func fromBSON() cli.Command {
...
 }
+
+func fromCSV() cli.Command {
+	return cli.Command{
+		Name:  "csv",
+		Usage: "write data from a CSV source to an FTDC file",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  input,
+				Usage: "write BSON data from this file (default: stdin)",
```
Comment (julianedwards) [inline operations/ftdc.go]: `source CSV data file`?

File: operations/ftdc.go [hunk 3]
```diff
@@ -278,15 +311,131 @@ func fromBSON() cli.Command {
...
+			},
+			cli.StringFlag{
+				Name:  output,
+				Usage: "write BSON data in FTDC format to this file",
```
Comment (julianedwards) [inline operations/ftdc.go]: CSV data*

File: operations/ftdc.go [hunk 4]
```diff
@@ -250,23 +268,38 @@ func fromBSON() cli.Command {
...
-			_, cancel := context.WithCancel(context.Background())
-			defer cancel()
-
 			bsonPath := c.String(input)
-			ftdcPrefix := c.String(output)
+			ftdcPath := c.String(output)
 			maxCount := c.Int(maxCount)
 
+			// access the source file
+			//
```
[the remark this answers is not in the document]
Comment (tychoish) [reply operations/ftdc.go]: I think it makes these comments easier to read

**Pull request merged**